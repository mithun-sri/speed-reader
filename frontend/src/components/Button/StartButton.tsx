import { Box, IconButton } from "@mui/material";
import { useEffect, useState } from "react";
import { useGameScreenContext } from "../../views/GameScreen/GameScreen";

const StartButton = () => {
  const { incrementCurrentStage } = useGameScreenContext();
  const [fontSize, setFontSize] = useState(calculateFontSize());

  useEffect(() => {
    function handleResize() {
      setFontSize(calculateFontSize());
    }
    window.addEventListener("resize", handleResize);
    return () => {
      window.removeEventListener("resize", handleResize);
    };
  }, []);

  function calculateFontSize() {
    const windowWidth = window.innerWidth;
    const minSize = 12;
    const maxSize = 36;
    const calculatedSize = Math.min(
      maxSize,
      Math.max(minSize, windowWidth / 15),
    );
    return calculatedSize;
  }

  return (
    <IconButton
      onClick={() => {
        incrementCurrentStage();
      }}
      sx={{
        fontFamily: "JetBrains Mono, monospace",
        color: "#FFFFFF",
      }}
    >
      <Box
        sx={{
          border: "10px solid #646669",
          borderRadius: "30px",
          background: "#E2B714",
          padding: "7px 40px 7px 40px",
          fontWeight: "bolder",
          fontSize: fontSize * 1,
        }}
      >
        Start
      </Box>
    </IconButton>
  );
};

export default StartButton;
