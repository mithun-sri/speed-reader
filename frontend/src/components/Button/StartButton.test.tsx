import renderer, { ReactTestRendererJSON } from "react-test-renderer";
import StartButton from "./StartButton";

it("Matches snapshot before/after click", () => {
  const component = renderer.create(<StartButton />);
  const tree1 = component.toJSON() as ReactTestRendererJSON;
  expect(tree1).toMatchSnapshot();

  renderer.act(() => {
    tree1.props.onClick();
  });
  const tree2 = component.toJSON() as ReactTestRendererJSON;
  expect(tree2).toMatchSnapshot();
});
