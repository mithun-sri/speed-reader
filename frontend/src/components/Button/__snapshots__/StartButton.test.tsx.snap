// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot before/after click 1`] = `
<button
  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-11jexo3-MuiButtonBase-root-MuiIconButton-root"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <div
    className="MuiBox-root css-93ovt4"
  >
    Start
  </div>
</button>
`;

exports[`Matches snapshot before/after click 2`] = `
<button
  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-11jexo3-MuiButtonBase-root-MuiIconButton-root"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <div
    className="MuiBox-root css-93ovt4"
  >
    Start
  </div>
</button>
`;
